pkg/events/events.go

```diff
@@ -0,0 +1,59 @@
+package events
+
+import (
+	"encoding/json"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+)
+
+// type EventType string
+
+type EventData struct {
+	URL           string `json:"url"`
+	CommitId      string `json:"commitId"`
+	RepositoryURL string `json:"repositoryUrl"`
+}
+
+type Event struct {
+	cloudEvent  cloudevents.Event
+	eventType   string
+	eventSource string
+}
+
+func NewEvent(eventType, eventSource string) Event {
+	return Event{
+		eventType:   eventType,
+		eventSource: eventSource,
+	}
+}
+
+func (e Event) CreateWithJSONData(data string, opts ...Option) Event {
+	return e.Create(data, opts...)
+}
+
+func (e Event) Create(data any, opts ...Option) Event {
+	e.cloudEvent = cloudevents.NewEvent("1.0")
+	// set default values
+	e.cloudEvent.SetID(uuid.New().String())
+	e.cloudEvent.SetType(e.eventType)
+	e.cloudEvent.SetTime(time.Now())
```

Comment:
Maybe we need to use UTC

```suggestion
	e.cloudEvent.SetTime(time.Now().UTC())
```

Reply:
This need to be ISO Timestamp!

Reply:
time.Now().Format(time.RFC3339) seems to be the way to go, as RFC3339 is a subset of ISO